### PR TITLE
Address secrel by adding the latest version of sqlfluff and flyway

### PR DESCRIPTION
**Pull request by greene858**
## What was the problem?
Older version of sqlfluff used in flyway was irritating secrel

Associated tickets or Slack threads:
#1772 
- https://dsva.slack.com/archives/C04CA47HV96/p1689825296169739

## How does this fix it?
In db-init Dockerfile, explicitly call pip to install version 2.1.2 of sqlfluff

## Note:
This will require an update if/when sqlfluff releases a newer version.

### CODE REVIEW

**Review by msnwatson — COMMENTED** [review]
can you link a successful SecRel run?

**Comment by greene858** [comment]
SecRel success: https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/5649956902  
Note: this was run after the suppression of the sqlfluff version was removed.

File: db-init/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,7 @@
-FROM flyway/flyway:9.20-alpine
+FROM flyway/flyway:9.21-alpine
 
 # hadolint ignore=DL3018
-RUN apk update && apk upgrade && apk --no-cache add libcrypto3 libssl3 openssl  && rm -rf /var/cache/apk/*
-
+RUN apk update && apk upgrade && apk --no-cache add libcrypto3 libssl3 openssl  && rm -rf /var/cache/apk/* && pip3 install --no-cache-dir sqlfluff==2.1.2
```
Comment (yoomlam) [inline db-init/Dockerfile:4]: Separate apk and pip installs:
```suggestion
RUN apk update && apk upgrade && apk --no-cache add libcrypto3 libssl3 openssl  && rm -rf /var/cache/apk/* 
RUN pip3 install --no-cache-dir sqlfluff==2.1.2
```

**Review by yoomlam — APPROVED** [review]

**Review by msnwatson — APPROVED** [review]
lgtm, once Yoom's comments are in

File: db-init/Dockerfile (hunk 1 above)
Comment (greene858) [reply db-init/Dockerfile]: Done

**Pull request merged**